src/app/modules/shared-components/components/add-content/add-content.component.ts

```diff
@@ -58,8 +59,15 @@ export class AddContentComponent<Type, SearchedValue extends AddedContent<Type>>
       existingTitleControl.pipe(
         map(value => value.trim()),
         filter(value => this.checkLength(value)),
-      ).subscribe(value => {
-        this.search.emit(value);
+        switchMap(value => merge(
+          of(fetchingState()),
+          of(value).pipe(delay(300), switchMap(value =>
+            (this.searchFunction ? this.searchFunction(value).pipe(map(readyState)) : throwError(new Error('no search function'))),
```

Comment:
this.searchFunction not set is something which is unexpected, so don't pollute the code with that, just throw at the beginning of "init" if not set and put it in another variable (which will not be optional) that you can use here.

Reply:
fixed